File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        if(nums.size() == 0)
        return 0;
        
        sort(nums.begin(),nums.end());
        int temp =1, ans =0;
        int prev = nums[0];
        unordered_set<int> a;
        for(int i=0;i<nums.size();i++)
        {
            if(a.find(nums[i])== a.end()){
                a.insert(nums[i]);
                if(prev + 1 == nums[i]){
                    temp++;
                }
                else{
                    if(temp > ans)
                    ans = temp;
                    temp =1;
                }
            }

            prev = nums[i];
        }
        if(temp > ans)
        ans = temp;
        return ans;
    }  
};

File: Data Structures & Algorithms/house-robber-ii/submission-1.java
class Solution {
    public int rob(int[] nums) {
        int[] dp = new int[2];
        int n = nums.length;
        int ans = nums[0];
        for(int i=0; i< n - 1;i++){
            int temp = dp[1];
            dp[1] = dp[0] + nums[i];
            dp[0] = temp;
            dp[1] = Math.max(dp[1], dp[0]);
            ans = Math.max(ans, dp[1]);
        }    
        
        dp = new int[2];
        for(int i=1; i< n;i++){
            int temp = dp[1];
            dp[1] = dp[0] + nums[i];
            dp[0] = temp;
            dp[1] = Math.max(dp[1], dp[0]);
            ans = Math.max(ans, dp[1]);
        } 

        return ans;                                                                                                                                                                                                                                           
    }
}
